### PR TITLE
Weth Reserve Implementation

### PR DESCRIPTION
**Pull request by talbaneth**
Includes testing.
Also includes an addition in helper to make throw catching more neat.

### CODE REVIEW

File: contracts/wethContracts/KyberWethReserve.sol [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        require(getDecimals(_wethToken) == COMMON_DECIMALS);
+
+        kyberNetwork = _kyberNetwork;
+        wethToken = _wethToken;
+        tradeEnabled = true;
+        admin = _admin;
+    }
+
+    function() public payable {
+        require(msg.sender == address(wethToken));
```
Comment (yaronvel) [inline contracts/wethContracts/KyberWethReserve.sol]: network can also send ether right?

File: contracts/wethContracts/KyberWethReserve.sol [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+    function getConversionRate(ERC20 src, ERC20 dest, uint srcQty, uint blockNumber) public view returns(uint) {
+
+        srcQty;
+        blockNumber;
+
+        if (!tradeEnabled) return 0;
+        if ((wethToken != src) && (wethToken != dest)) return 0;
+        if ((ETH_TOKEN_ADDRESS != src) && (ETH_TOKEN_ADDRESS != dest)) return 0;
+
```
Comment (yaronvel) [inline contracts/wethContracts/KyberWethReserve.sol]: should also block ETH to ETH and WETH to WETH.

File: contracts/wethContracts/KyberWethReserve.sol [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        uint srcAmount,
+        ERC20 destToken,
+        address destAddress,
+        uint conversionRate,
+        bool validate
+    )
+        internal
+        returns(bool)
+    {
+        require((ETH_TOKEN_ADDRESS == srcToken) || (ETH_TOKEN_ADDRESS == destToken));
```
Comment (yaronvel) [inline contracts/wethContracts/KyberWethReserve.sol]: should also block ETH to ETH and WETH to WETH

**Comment by ilanDoron** [comment]
Code is same as deployed code.

**Pull request merged**